JOSS compliance (#67)
CDCgov/nbbp: paper/paper.md

# Summary

During an infectious disease outbreak, each infected person, or "case," infects a certain number of other people, often zero.

> In a Galton-Watson branching process model of an outbreak, the number of onward infections per infector is drawn from a statistical distribution, such as the negative binomial, parameterised by the effective reproduction number $R$ and a concentration parameter $k$, such that the distribution has mean $R$ and variance $R + R^2/k$.

Reviewer: ```suggestion
In a branching process model of an outbreak, the number of onward infections per infector is drawn from some distribution, commonly the negative binomial, which is parameterised by the effective reproduction number $R$ and concentration parameter $k$, such that the distribution has mean $R$ and variance $R + R^2/k$.
```

Mostly not actually important wordsmithing, does reduce word count by net 3 and I find a little clearer this way.

Mostly I object to putting Galton-Watson in unnecessarily.

Reviewer: Clearer as in "does not imply all distributions used have these parameters, just the NB"

Inferring the parameters of that distribution is important for forecasting the future of outbreaks and for determining what interventions are most appropriate.
`nbbp` is an [R](https://www.r-project.org/) [@r2021r] package for Bayesian inference of negative binomial branching process parameters using final outbreak size data [@blumberg2013inference; @nishiura2012estimating].

# Statement of need

Ideally, investigations of infectious disease outbreaks would yield a complete transmission tree, showing who infected whom.
In practice, we often only have the final size of each outbreak.

> Methods to infer negative binomial branching process parameters from final sizes can suffer problems when outbreaks are small, when there are small numbers of outbreaks, and when outbreaks are large.

Reviewer: Making sure I'm following, we're saying (nicely compactly): all-singletons (and similar) are hard, small datasets are hard, and non-extinct-type-things are hard?

Reviewer: ```suggestion
Methods to infer negative binomial branching process parameters from final sizes can suffer problems when outbreaks are small, when there are small numbers of outbreaks, and when outbreaks are exceptionally large.
```

> Accurate analysis of small datasets requires Bayesian approaches because standard statistical guarantees like unbiasedness may not be useful when estimator variance are very large, and frequentist confidence intervals based on large-sample theory may be misleading or invalid.

Reviewer: ```suggestion
Accurate analysis of small datasets benefits from Bayesian approaches because standard statistical guarantees like unbiasedness may not be useful when estimator variance are very large, and frequentist confidence intervals based on large-sample theory may be misleading or invalid.
```

Everything has some problem here, and while there are lots of reasons to like Bayes, I think we'd do best to let the problems we talk about do the convincing.

Reviewer: I'm... not liking some of the rest of this sentence either. Maybe I ought to just open a PR into this with my preferred takes?

These distortions can affect probabilistic judgments on questions like "Is $R \leq 1$ or not?" which can be crucial for public health decision-making.

> Analysis of large outbreaks requires explicit censoring and probabilistic conditioning to avoid drawing misleading conclusions.

Reviewer: Analysis of large outbreaks requires _care_ and _caution_.

1. You can either:
    1. Censor it
    2. Say "it's non-extinct"
2. You should _not_:
    1. Condition on extinction

We're missing 1.ii and seem to contradict 2.i?

> Galton-Watson branching processes assume that the offspring distribution is constant throughout every outbreak, which can be a good model of epidemic dynamics when outbreaks are new and small, but is inappropriate for large outbreaks, since these processes inevitably ends in:

Reviewer: ```suggestion
Branching processes assume that the offspring distribution is constant throughout every outbreak, which can be a good model of epidemic dynamics when outbreaks are new and small, but is inappropriate for large outbreaks, since these processes inevitably ends in:
```

I'm going to need serious convincing that Galton-Watson is worth saying in general, and especially when we need to keep the paper trim.

2. flexible handling of exceptionally large outbreaks.

# Software design

## Software ecosystem

`nbbp` uses [Stan](https://mc-Stan.org/) [@stan2026stan] for statistical inference, interfacing with R using [`Rstan`](https://mc-Stan.org/rstan/articles/rstan.html) [@stan2025rstan] and [`rstantools`](https://mc-Stan.org/rstantools/) [@gabry2026rstantools]. By providing standard `rstan` outputs, `nbbp` can be integrated into larger epidemiological analyses using this mature Bayesian software ecosystem. `nbbp`'s design is modular, enabling the addition of new models (e.g., extensions to time-series or geospatial modeling) that leveraging the underlying Stan codebase with minimal architecture changes. The package is extensively tested via `testthat`.

## Support for censoring, partially observed outbreaks, and non-extinction

For inference, `nbbp` accommodates a wide variety of observation processes for finite chain sizes as well as flexible handling for exceptionally large outbreaks.
In total, it ecognizes four kinds of observed chain sizes as input:

> 1. **Completely observed and extinct**, optionally of a minimum size. The exact number $c$ of cases in the outbreak is known, and outbreaks of size at least $C$ are observed. If all "outbreaks", including single infections, are reported, then $C = 1$. Otherwise, $C$ is the minimum outbreak size required for an outbreak to appear in the dataset.

Reviewer: Should we reference Nishiura here since it's now been mentioned above?

1. **Censored**. The number of cases is $c \geq C$.
1. **Partially observed and extinct**. Each case in the outbreak has an independent probability $p$ of being observed, and $c$ cases are observed.
1. An **explosion**, that is, an outbreak that grew (or would have grown) to infinite size.

> Accordingly, `nbbp` implements the log-likelihood for a given dataset as:

Reviewer: ```suggestion
Accordingly, `nbbp`'s implementation of the log-likelihood for a given dataset can be written as:
```

Exceedingly pedantic, and not particularly important, because no one would actually put the -0 terms in for Pr( | C >= 1).

## User friendlieness

`nbbp` aims to be useful to practitioners by providing:

- **Likelihood surface visualization**, aiding in model diagnostics and parameter exploration, and showing clearly the value of Bayesian over frequentist inference for certain datasets.
- **Package data**: six exemplar datasets of final size outbreak sizes, covering a range of data and parameter regimes [@blumberg2014detecting; @nigel2004assessment; @king2004measles; @cauchemez2014middle].
- **Simulation-tested settings**. Performance of Bayesian estimators using the default prior settings have been tested in simulation, including grid-based simulation study and a test of the calibration of posterior credible intervals, helping users understand expected performance over a range of parameter space.
- **Maximum likelihood inference** (MLE). `nbbp` was designed in part because of MLE's sensitivity to pathologies common to the likelihood surfaces of these types of datasets. `nbbp` provides some MLE methods, with enhanced numerical safeguards. Nevertheless, MLE results should be used only for data exploration and for comparison to Bayesian results.

> # Research impact

Reviewer: I feel like we're liable to provoke a knee-jerk reaction of "this is speculative" from reviewers.

Reviewer: In #68 I tried to leave what I thought safe and moved the rest to the "Statement of need"
